fe/fe-core/src/main/java/com/starrocks/connector/jdbc/MysqlSchemaResolver.java

```diff
@@ -0,0 +1,91 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.connector.jdbc;
+
+import com.google.common.collect.ImmutableSet;
+import com.starrocks.catalog.PrimitiveType;
+import com.starrocks.catalog.ScalarType;
+import com.starrocks.catalog.Type;
+import com.starrocks.common.DdlException;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.Collection;
+
+import static java.lang.Math.max;
+
+public class MysqlSchemaResolver extends JDBCSchemaResolver {
+
+    @Override
+    public Collection<String> listSchemas(Connection connection) throws DdlException {
+        try (ResultSet resultSet = connection.getMetaData().getCatalogs()) {
+            ImmutableSet.Builder<String> schemaNames = ImmutableSet.builder();
+            while (resultSet.next()) {
+                String schemaName = resultSet.getString("TABLE_CAT");
+                // skip internal schemas
+                if (!schemaName.equalsIgnoreCase("information_schema") &&
+                        !schemaName.equalsIgnoreCase("mysql") &&
+                        !schemaName.equalsIgnoreCase("performance_schema") &&
+                        !schemaName.equalsIgnoreCase("sys")) {
+                    schemaNames.add(schemaName);
+                }
+            }
+            return schemaNames.build();
+        } catch (SQLException e) {
+            throw new DdlException(e.getMessage());
+        }
+    }
+
+    @Override
+    public Type convertColumnType(int dataType, String typeName, int columnSize, int digits) {
+        PrimitiveType primitiveType;
+        switch (dataType) {
+            case Types.BOOLEAN:
+                primitiveType = PrimitiveType.BOOLEAN;
+                break;
+            case Types.TINYINT:
+                primitiveType = PrimitiveType.TINYINT;
+                break;
+            case Types.SMALLINT:
+                primitiveType = PrimitiveType.SMALLINT;
+                break;
+            case Types.INTEGER:
```

Comment:
MySQL has a MEDIUMINT type, does this correspond to INTEGER?

Reply:
Yes，Mysql MEDIUMINT correspond to INTEGER in JDBC

Reply:
Yes，Mysql MEDIUMINT correspond to INTEGER in JDBC